src/bpftrace.h

```diff
@@ -190,7 +190,7 @@ class BPFtrace {
   const FuncsModulesMap &get_traceable_funcs() const;
   KConfig kconfig;
   std::vector<std::unique_ptr<AttachedProbe>> attached_probes_;
-  std::unique_ptr<AttachedProbe> sigusr1_attached_probe_;
+  std::optional<int> sigusr1_prog_fd_;
```

Comment:
using optional instead of initializing the `prog_fd` with a value because `bpf_prog_test_run_opts` will raise an error for errored fds (similiar to BEGIN/END)